routes/api/contacts.js

```diff
@@ -1,25 +1,84 @@
-const express = require('express')
+const {
+  listContacts,
+  getContactById,
+  addContact,
+  removeContact,
+  updateContact,
+} = require("../../models/contacts");
 
-const router = express.Router()
+const express = require("express");
 
-router.get('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+const Joi = require("joi");
+const schema = Joi.object({
+  name: Joi.string().alphanum().min(2).max(30),
+  email: Joi.string().email({ minDomainSegments: 2, multiple: true }),
+  phone: Joi.string().alphanum().min(9).max(15),
+});
 
-router.get('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+const router = express.Router();
 
-router.post('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/", async (req, res, next) => {
+  try {
+    const contacts = await listContacts();
+    res.json(contacts);
+  } catch (error) {
+    console.log(error);
+  }
+});
 
-router.delete('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/:contactId", async (req, res, next) => {
+  try {
+    const id = req.params.contactId;
+    const searchedContact = await getContactById(id);
+    searchedContact
+      ? res.status(200).json(searchedContact)
+      : res.status(404).json({ message: "Not found" });
+  } catch (error) {
+    console.log(error);
+  }
+});
 
-router.put('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.post("/", async (req, res, next) => {
+  try {
+    const validateResult = schema.validate(req.body);
+    if (validateResult.error || !req.body)
+      return res.status(400).json({ message: "missing required name - field" });
```

Comment:
To nie zawsze jest prawdą? Bo co jeżeli ktoś umieści Ci number a ty chcesz string? Pole będzie obecne ale dane nei będą właściwe i bedzie inna wiadomość. Wykorzystaj wiadomość z joi

Reply:
Poprawiłem przekazywanie informacji o błędzie dla usera (info z Joi) oraz dodałem next do bloków catch